cdap-ui/app/cdap/components/HeaderNavbarList/index.js

```diff
@@ -13,28 +13,58 @@
  * License for the specific language governing permissions and limitations under
  * the License.
  */
-import React, {PropTypes} from 'react';
+import React, {Component, PropTypes} from 'react';
 import {Link} from 'react-router';
+import Store from '../../services/store/store.js';
 
-export default function HeaderNavbarList({list}) {
-  return (
-    <ul className="navbar-list">
-      {
-        list.map((navbaritem, index) => {
-          return (
-            <li key={index}>
-              <Link
-                to={navbaritem.linkTo}
-                activeClassName="active"
-              >
-                {navbaritem.title}
-              </Link>
-            </li>
-          );
-        })
-      }
-    </ul>
-  );
+export default class HeaderNavbarList extends Component {
```

Comment:
We don't need to use es6 class here. We should use `react-redux`'s `connect` function. It automatically takes care of subscribing, binding it to properties, updating the state and unsubscribing from the store on unMounting the component. The boilerplate of doing these are already in there.